### PR TITLE
New handlebars titlecase helper

### LINKED ISSUE #517: [NV-370] - Templates - Handlebars - new titlecase helper
## Why?

When triggering an event on the `@novu/api`a message is compiled using handlebars. We want to allow the user to specify some handlebars helpers when creating a message template in the UI. 

## What?

A new handlebars helper read more about it ([https://handlebarsjs.com/guide/#custom-helpers](https://handlebarsjs.com/guide/#custom-helpers)) called `titlecase` when applied, the variable content will be converted to a title cased version of it. For example:

* `hello world` > `Hello World`
* `hello World` > `Hello World`

### Example of usage within a template

```
Hello !
```

## Done Criteria

* added a test for those use cases`apps/api/src/app/content-templates/usecases/compile-template/compile-template.spec.ts`

### PR DESCRIPTION
**Pull request by AlexVCS**
- **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
- Add new handlebars helper for custom message creation

- **What is the current behavior?** (You can also link to an open issue here)
Closes #517 

- **What is the new behavior (if this is a feature change)?**

- **Other information**:

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -12,5 +12,8 @@
   },
   "cSpell.words": [
     "mantine"
-  ]
+  ],
+  "[typescript]": {
+    "editor.defaultFormatter": "esbenp.prettier-vscode"
+  }
```
Comment (ainouzgali) [inline .vscode/settings.json:18]: Could you please revert this as well? :)

File: apps/api/src/app/content-templates/usecases/compile-template/compile-template.spec.ts [hunk 2]
```diff
@@ -5,6 +5,7 @@ import { SharedModule } from '../../../shared/shared.module';
 import { ContentTemplatesModule } from '../../content-templates.module';
 import { CompileTemplate } from './compile-template.usecase';
 import { CompileTemplateCommand } from './compile-template.command';
+import { functionsIn } from 'lodash';
```
Comment (ainouzgali) [inline apps/api/src/app/content-templates/usecases/compile-template/compile-template.spec.ts:8]: I think this one might have been added by accident

**Review by ainouzgali — COMMENTED** [review]
@AlexVCS  looks great! Thank you so much.🚀
Just please notice what needs to be removed.

File: apps/api/src/app/content-templates/usecases/compile-template/compile-template.usecase.ts [hunk 3]
```diff
@@ -8,6 +8,21 @@ Handlebars.registerHelper('equals', function (arg1, arg2, options) {
   return arg1 == arg2 ? options.fn(this) : options.inverse(this);
 });
 
+Handlebars.registerHelper('titlecase', function (string) {
```
Comment (scopsy) [inline apps/api/src/app/content-templates/usecases/compile-template/compile-template.usecase.ts:11]: While `string` is not a reserved word, some ide's might interpret this with wrong formatting. What do you think about switching this to str? or value? 
```suggestion
Handlebars.registerHelper('titlecase', function (value) {
```

File: apps/api/src/app/content-templates/usecases/compile-template/compile-template.usecase.ts [hunk 4]
```diff
@@ -8,6 +8,21 @@ Handlebars.registerHelper('equals', function (arg1, arg2, options) {
...
+  return string
```
Comment (scopsy) [inline apps/api/src/app/content-templates/usecases/compile-template/compile-template.usecase.ts:12]:
```suggestion
  return value?
```

To make sure we are not failing on null or falsy values here.

File: apps/api/src/app/content-templates/usecases/compile-template/compile-template.usecase.ts [hunk 5]
```diff
@@ -8,6 +8,21 @@ Handlebars.registerHelper('equals', function (arg1, arg2, options) {
...
+    .split(' ')
+    .map((letter) => letter.charAt(0).toUpperCase() + letter.slice(1).toLowerCase())
+    .join(' ');
+});
+
+Handlebars.registerHelper('uppercase', function (string) {
+  return string.toUpperCase();
+});
```
Comment (scopsy) [inline apps/api/src/app/content-templates/usecases/compile-template/compile-template.usecase.ts:20]:
```suggestion
Handlebars.registerHelper('uppercase', function (value) {
  return value?.toUpperCase();
});
```

File: apps/api/src/app/content-templates/usecases/compile-template/compile-template.usecase.ts [hunk 6]
```diff
@@ -8,6 +8,21 @@ Handlebars.registerHelper('equals', function (arg1, arg2, options) {
...
+
+Handlebars.registerHelper('lowercase', function (string) {
+  return string.toLowerCase();
+});
```
Comment (scopsy) [inline apps/api/src/app/content-templates/usecases/compile-template/compile-template.usecase.ts:24]:
```suggestion
Handlebars.registerHelper('lowercase', function (value) {
  return value?.toLowerCase();
});
```

**Review by scopsy — COMMENTED** [review]
@AlexVCS Looks great!! Added a few suggestions and a null or falsy value handling aswel. Let me know what you think 🎉

**Comment by scopsy** [comment]
@AlexVCS You are the best! Merging this soon.

**Review by scopsy — APPROVED** [review]

**Pull request merged**